tests/rptest/tests/acls_test.py

```diff
@@ -307,48 +349,22 @@ def test_mtls_principal(self, rules=None, fail=False):
                              authn_method="mtls_identity",
                              principal_mapping_rules=rules)
 
-        # run a few times for good health
-        for _ in range(5):
-            try:
-                self.get_client("cluster_describe").acl_list()
-                assert not fail, "list acls should have failed"
-            except ClusterAuthorizationError:
-                assert fail, "list acls should have succeeded"
+        self.check_permissions(pass_w_base_user=False,
+                               pass_w_cluster_user=not fail,
+                               pass_w_super_user=not fail,
+                               err_msg='check_permissions failed')
 
     @cluster(num_nodes=3)
     @parametrize(authn_method='sasl')
     @parametrize(authn_method='mtls_identity')
     def test_security_feature_migration(self, authn_method: str):
-        def check_perms():
-            try:
-                self.get_client("base").acl_list()
-                self.logger.debug('list acls should have failed for base user')
-                return False
```

Comment:
this was just added in the previous commit. that suggests there is a way to split up this commit to avoid the adding / immediate removal?

Reply:
Yes I can avoid the immediate removal. I'll re-organize the commits.